app/javascript/components/Audience/CustomersPage.tsx

```diff
@@ -685,25 +731,127 @@ const CustomerDrawer = ({
+
+  const workflowIdForQuery = (workflowId: string | undefined) => (workflowId === "" ? undefined : workflowId);
+
+  const isProcessingMissedPostsJobForPurchase = (purchaseExternalId: string) =>
+    Array.from(missedPostsJobsInProcessRef.current).some((jobKey) => {
+      const job = cast<{ customerId: string; workflowId?: string }>(JSON.parse(jobKey));
+      return job.customerId === purchaseExternalId;
     });
-    getCustomerEmails(customer.id).then(setEmails, (e: unknown) => {
+
+  const isProcessingAllOrSpecificJob = (purchaseExternalId: string, workflowId?: string) => {
+    if (workflowId === undefined) {
+      return isProcessingMissedPostsJobForPurchase(purchaseExternalId);
+    }
+
+    return (
+      missedPostsJobsInProcessRef.current.has(JSON.stringify({ customerId: purchaseExternalId, workflowId })) ||
+      missedPostsJobsInProcessRef.current.has(JSON.stringify({ customerId: purchaseExternalId }))
+    );
+  };
+
+  const [processingJobForWorkflow, setProcessingJobForWorkflow] = React.useState(
+    isProcessingAllOrSpecificJob(customer.id, workflowIdForQuery(selectedWorkflowId)),
+  );
+
+  const fetchAndDisplayMissedPostsByWorkflowId = (workflowId: string) => {
+    router.reload({
+      data: {
+        purchase_id: customer.id,
+        workflow_id: workflowIdForQuery(workflowId),
+      },
+      only: ["missed_posts"],
+      preserveUrl: true,
+      onStart: () => setIsLoadingMissedPosts(true),
+      onFinish: () => setIsLoadingMissedPosts(false),
+    });
+    setSelectedWorkflowId(workflowId);
+    setProcessingJobForWorkflow(isProcessingAllOrSpecificJob(customer.id, workflowIdForQuery(workflowId)));
+  };
+
+  const resendAllMissedPosts = async () => {
+    if (!cable) return;
+
+    const resendJobWorkflowId = workflowIdForQuery(selectedWorkflowId);
+    const resendJobKeyLocal = JSON.stringify({ customerId: customer.id, workflowId: resendJobWorkflowId });
+
+    let failedToSubscribeToCustomersChannel = false;
+
+    try {
+      if (!customersChannelRef.current) {
+        customersChannelRef.current = cable.subscribeTo(CUSTOMERS_CHANNEL_NAME, { purchase_id: customer.id });
+
+        customersChannelRef.current.on("message", (packet) => {
+          if (!is<IncomingCustomersChannelMessage>(packet)) return;
+          const resendJobKeyFromPacket = JSON.stringify({
+            customerId: packet.purchase_id,
+            workflowId: packet.workflow_id,
+          });
+
+          if (!missedPostsJobsInProcessRef.current.has(resendJobKeyFromPacket)) return;
+
+          missedPostsJobsInProcessRef.current.delete(resendJobKeyFromPacket);
+
+          switch (packet.type) {
+            case "missed_posts_job_complete":
+              showAlert(packet.message, "success");
+              break;
+            case "missed_posts_job_failed":
+              showAlert(packet.message, "error");
+              break;
+          }
+        });
```

Comment:
To handle updates when the drawer is closed, we create a subscription for it once inside `CustomerDrawer`. 

I'm splitting up the logic to clear the job from store over here as well to simplify re-conciliation logic down below